src/protocol.h

```diff
@@ -354,34 +354,45 @@ static inline bool MayHaveUsefulAddressDB(ServiceFlags services)
     return (services & NODE_NETWORK) || (services & NODE_NETWORK_LIMITED);
 }
 
+enum class AddrSerialization {
+    DISK,
+    NETWORK_NOTIME,
+    NETWORK_WITHTIME,
+};
+
 /** A CService with information about it as peer */
 class CAddress : public CService
 {
     static constexpr uint32_t TIME_INIT{100000000};
 
+    /** The disk serialization for CAddress includes a version number.
+     *  Traditionally the number CLIENT_VERSION was used, but it has since
+     *  been disentangled from it. */
+    static constexpr uint32_t DISK_VERSION{210000};
+
 public:
     CAddress() : CService{} {};
     explicit CAddress(CService ipIn, ServiceFlags nServicesIn) : CService{ipIn}, nServices{nServicesIn} {};
 
-    SERIALIZE_METHODS(CAddress, obj)
+    SERIALIZE_METHODS_PARAMS(CAddress, obj, AddrSerialization, fmt)
     {
         SER_READ(obj, obj.nTime = TIME_INIT);
-        int nVersion = s.GetVersion();
-        if (s.GetType() & SER_DISK) {
-            READWRITE(nVersion);
+        switch (fmt) {
+        case AddrSerialization::DISK: {
+            uint32_t disk_version = DISK_VERSION;
+            READWRITE(disk_version);
+            READWRITE(obj.nTime);
+            break;
         }
-        if ((s.GetType() & SER_DISK) ||
-            (nVersion != INIT_PROTO_VERSION && !(s.GetType() & SER_GETHASH))) {
+        case AddrSerialization::NETWORK_WITHTIME:
+            READWRITE(obj.nTime);
+            break;
+        case AddrSerialization::NETWORK_NOTIME:
             // The only time we serialize a CAddress object without nTime is in
```

Comment:
Thinking about it a bit more, I actually think this whole comment can be removed. It made sense when the serialization was based on a protocol version that we don't support anywhere else (and was added to document something that caused confusion for reviewers: https://github.com/bitcoin/bitcoin/pull/14033#issuecomment-652456443), but when the serialization is based on a parameter that we pass in explicitly, it's easy enough to grep for that parameter to see where it's used.

Reply:
Done.